### PR TITLE
Use dune and define profile "gprof" for profiling

### PR DESCRIPTION
**Pull request by lindig**
Switch to to Dune and define a profile for profiling with gprof while
keeping jbuild files for now. Since Dune has stricter warning flags in
its dev profile, we have to use the release profile to avoid braking the
build until this is cleaned up. Switching to a different profile is easy
from the command line:

  make PROFILE=dev

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

### CODE REVIEW

File: dune [hunk 1]
```diff
@@ -0,0 +1,20 @@
+(env
+ (gprof
+  (ocamlopt_flags
+   (:standard -g -p -w -39)
```
Comment (edwintorok) [inline dune]: I think `-g` is there by default, otherwise backtraces wouldn't work, but it doesn't hurt to be explicit.

File: dune [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  )
+ )
+ (dev
+  (flags
+   (:standard -g -w -39)
+  )
+ )
+ (release
+  (flags
+   (:standard -w -39)
```
Comment (mseri) [inline dune]: Are you sure you want to get rid of `g` here? The code may become undebuggable

Comment (edwintorok) [reply dune]: -g should be part of :standard

File: dune (hunk 1 above)
Comment (lindig) [reply dune]: According to the logs, we are still compiling with `-g`.

**Pull request merged**